bundles/org.openhab.binding.wlanthermo/src/main/java/org/openhab/binding/wlanthermo/internal/WlanThermoNanoHandler.java

```diff
@@ -0,0 +1,203 @@
+        } else {
+            if (wlanThermoNanoCommandHandler.setState(channelUID, command, data)) {
+                logger.debug("Data updated, pushing changes");
+                push();
+            } else {
+                logger.warn("Could not handle command of type {} for channel {}!", command.getClass().toGenericString(),
+                        channelUID.getId());
+            }
+        }
+    }
+
+    private void update() {
+        try {
+            // Update objects with data from device
+            String json = httpClient.GET(config.getUri("/data")).getContentAsString();
+            data = gson.fromJson(json, Data.class);
+            logger.debug("Received at /data: {}", json);
+            json = httpClient.GET(config.getUri("/settings")).getContentAsString();
+            settings = gson.fromJson(json, Settings.class);
+            logger.debug("Received at /settings: {}", json);
+
+            // Update channels
+            for (Channel channel : thing.getChannels()) {
+                State state = wlanThermoNanoCommandHandler.getState(channel.getUID(), data, settings);
+                if (state != null) {
+                    updateState(channel.getUID(), state);
+                } else {
+                    // if we could not obtain a state, try trigger instead
+                    String trigger = wlanThermoNanoCommandHandler.getTrigger(channel.getUID(), data);
+                    if (trigger != null) {
+                        triggerChannel(channel.getUID(), trigger);
+                    }
+                }
+            }
+        } catch (URISyntaxException | InterruptedException | ExecutionException | TimeoutException e) {
+            logger.debug("Update failed, checking connection", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Update failed, reconnecting...");
+            if (pollingScheduler != null) {
+                pollingScheduler.cancel(false);
+            }
+            for (Channel channel : thing.getChannels()) {
+                updateState(channel.getUID(), UnDefType.UNDEF);
+            }
+            checkConnection();
+        }
+    }
+
+    private void push() {
+        for (org.openhab.binding.wlanthermo.internal.api.nano.data.Channel c : data.getChannel()) {
+            try {
+                String json = gson.toJson(c);
+                logger.debug("Pushing: {}", json);
+                URI uri = config.getUri("/setchannels");
+                int status = httpClient.POST(uri).content(new StringContentProvider(json), "application/json")
+                        .timeout(5, TimeUnit.SECONDS).send().getStatus();
+                if (status == 401) {
+                    updateStatus(ThingStatus.ONLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            "No or wrong login credentials provided. Please configure username/password for write access to WlanThermo!");
+                    logger.warn(
```

Comment:
Not needed, thing status changes are logged anyway, If you want to log something, maybe log the content response at TRACE or DEBUG level (if it adds useful information),